web/src/settings_profile_fields.ts

```diff
@@ -652,51 +653,54 @@ export function do_populate_profile_fields(profile_fields_data: CustomProfileFie
     // We should only call this internally or from tests.
     const $profile_fields_table = $("#admin_profile_fields_table").expectOne();
 
-    $profile_fields_table.find("tr.profile-field-row").remove(); // Clear all rows.
-    $profile_fields_table.find("tr.profile-field-form").remove(); // Clear all rows.
     order = [];
-
     let display_in_profile_summary_fields_count = 0;
-    for (const profile_field of profile_fields_data) {
-        order.push(profile_field.id);
-        let choices: FieldChoice[] = [];
-        if (profile_field.field_data && profile_field.type === field_types.SELECT.id) {
-            const field_data = settings_components.select_field_data_schema.parse(
-                JSON.parse(profile_field.field_data),
-            );
-            choices = parse_field_choices_from_field_data(field_data);
-        }
-        const display_in_profile_summary = profile_field.display_in_profile_summary === true;
-        const required = profile_field.required;
-        $profile_fields_table.append(
-            $(
-                render_admin_profile_field_list({
-                    profile_field: {
-                        id: profile_field.id,
-                        name: profile_field.name,
-                        hint: profile_field.hint,
-                        type: field_type_id_to_string(profile_field.type),
-                        choices,
-                        is_select_field: profile_field.type === field_types.SELECT.id,
-                        is_external_account_field:
-                            profile_field.type === field_types.EXTERNAL_ACCOUNT.id,
-                        display_in_profile_summary,
-                        valid_to_display_in_summary: is_valid_to_display_in_summary(
-                            profile_field.type,
-                        ),
-                        required,
-                    },
-                    can_modify: current_user.is_admin,
-                    realm_default_external_accounts: realm.realm_default_external_accounts,
-                }),
-            ),
-        );
 
-        // Keeping counts of all display_in_profile_summary profile fields, to keep track.
-        if (display_in_profile_summary) {
-            display_in_profile_summary_fields_count += 1;
-        }
-    }
+    ListWidget.create($profile_fields_table, profile_fields_data, {
```

Comment:
Is ListWidget necessary for the issue to be resolved? If no have you discussed this change in CZO?

Also just a nitpick but I think refactor might be more suitable than migrate for the commit message.

Reply:
Sorry I forgot to link [CZO discussion](https://chat.zulip.org/#narrow/stream/6-frontend/topic/Placeholder.20for.20profile.20fields.20org.20setting.20table.20.2327250/near/1822106). I have added this link to the PR aswell.

Reply:
Yeah don't like "Migrate..." either, but I decided against "Refactor..." since that seems to imply there is no change in logic. But actually we are now using ListWidget instead of manually creating the rows.

 I think maybe "Use ListWidget..." might be more appropriate? What do you think @sanchi-t?

Reply:
Yeah makes sense. This PR LGTM.